sound/soc/sof/ipc4-pcm.c

```diff
@@ -215,8 +215,41 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	return 0;
 }
 
+static int sof_ipc4_pcm_setup(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm)
+{
+	struct snd_sof_pipeline_trigger_info *pipeline_info;
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	int stream;
+
+	for_each_pcm_streams(stream) {
+		pipeline_info = &spcm->stream[stream].pipeline_info;
+
+		/* allocate memory for max number of pipeline IDs */
+		pipeline_info->pipeline_list = kcalloc(ipc4_data->max_num_pipelines,
+						       sizeof(struct snd_sof_widget *),
+						       GFP_KERNEL);
+		if (!pipeline_info->pipeline_list)
+			return -ENOMEM;
```

Comment:
do we need an unwind loop to free-up the memory that's been allocated earlier in the loop?

Reply:
yes